src/webots/wren/WbCoordinateSystem.cpp

```diff
@@ -167,10 +167,11 @@ void WbCoordinateSystem::setVisible(bool b) {
 
 void WbCoordinateSystem::setOrientation(const WbQuaternion &quaternion) {
   float rotation[4];
-  WbRotation(quaternion).toFloatArray(rotation);
+  WbQuaternion adaptedQuaternion(WbQuaternion(0.5, -0.5, 0.5, 0.5) * quaternion);
```

Comment:
I am not sure whether there is a cleaner way

Reply:
Can't think of any either.

Reply:
Can't think of a better way either.